client/lib/domains/utils.js

```diff
@@ -56,4 +55,28 @@ function getDomainNameFromReceiptOrCart( receipt, cart ) {
 	return null;
 }
 
-export { getDomainNameFromReceiptOrCart, getDomainType };
+function getWpcomTldsCachedList() {
+	return require( './tlds/wpcom-tlds.json' );
```

Comment:
Would it be simpler to export this as a value, rather than through a function?

Reply:
Babel takes it down to a `require`, right? I was wanting a `require` specifically so the list gets cached after first read.

Reply:
I think it will end up as a require, but I don't think that that buys you much in the browser.

I spent some time digging into this, and I'm slightly more confused than when I started, and a little dismayed at how little I understand how our [section loading magic](https://github.com/Automattic/wp-calypso/blob/fa0f28c8bd6b2aa5cba422b9f452fa5e328091d2/server/bundler/loader.js) interacts with webpack to get the job done.

Nonetheless, I think it's like this: Webpack bundles up everything into one file unless you explicitly break it apart into chunks. You can see the different chunks in the build, e.g.:
![calypso_ _node_ _npm_manpath__users_jules__nvm_versions_node_v6_11_2_share_man__usr_local_share_man__usr_share_man__opt_x11_share_man__applications_xcode_app_contents_developer_usr_share_man__applications_xcode_app_contents_developer_toolch](https://user-images.githubusercontent.com/5952255/31657729-78fded78-b372-11e7-8783-2620563360dc.jpg)

So webpack is going to tag the file as a dependency and include it in the `domains` chunk, unless we explicitly set up a new chunk to delay that loading. I'm pretty sure that's where the sections come in - making that process easier.

Once the chunk is loaded, it's going to sit in the webpack bundle inert, waiting to be imported or loaded:
![checkout_ _julesauspremiumtest_ _wordpress_com](https://user-images.githubusercontent.com/5952255/31658074-5ffdf772-b373-11e7-920c-8757860ecb4f.jpg)

From my testing, the the imports do get evaluated before the require, so I think that the `require()` won't affect or cache the loading, but it will delay and cache the evaluation/parsing for the JSON.

I don't really know how long that takes, but it's probably premature optimization.